packages/frama-c/frama-c.27.0/opam

```diff
@@ -0,0 +1,181 @@
+- an analyzer based on abstract interpretation (Eva plug-in);
+- a program proof framework based on weakest precondition calculus (WP plug-in);
+- a program slicer (Slicing plug-in);
+- a tool for verification of temporal (LTL) properties (Aoraï plug-in);
+- a runtime verification tool (E-ACSL plug-in);
+- several tools for code base exploration and dependency analysis
+  (plug-ins From, Impact, Metrics, Occurrence, Scope, etc.).
+These plug-ins communicate between each other via the Frama-C API
+and via ACSL (ANSI/ISO C Specification Language) properties.
+"""
+maintainer: "frama-ci-bot@frama-c.com"
+authors: [
+  "Michele Alberti"
+  "Thibaud Antignac"
+  "Gergö Barany"
+  "Patrick Baudin"
+  "Thibaut Benjamin"
+  "Allan Blanchard"
+  "Lionel Blatter"
+  "François Bobot"
+  "Richard Bonichon"
+  "Quentin Bouillaguet"
+  "David Bühler"
+  "Zakaria Chihani"
+  "Loïc Correnson"
+  "Julien Crétin"
+  "Pascal Cuoq"
+  "Zaynah Dargaye"
+  "Basile Desloges"
+  "Jean-Christophe Filliâtre"
+  "Philippe Herrmann"
+  "Maxime Jacquemin"
+  "Florent Kirchner"
+  "Alexander Kogtenkov"
+  "Tristan Le Gall"
+  "Jean-Christophe Léchenet"
+  "Matthieu Lemerre"
+  "Dara Ly"
+  "David Maison"
+  "Claude Marché"
+  "André Maroneze"
+  "Thibault Martin"
+  "Fonenantsoa Maurica"
+  "Melody Méaulle"
+  "Benjamin Monate"
+  "Yannick Moy"
+  "Anne Pacalet"
+  "Valentin Perrelle"
+  "Guillaume Petiot"
+  "Dario Pinto"
+  "Virgile Prevosto"
+  "Armand Puccetti"
+  "Félix Ridoux"
+  "Virgile Robles"
+  "Muriel Roger"
+  "Julien Signoles"
+  "Nicolas Stouls"
+  "Kostyantyn Vorobyov"
+  "Boris Yakobowski"
+]
+homepage: "https://frama-c.com/"
+license: "LGPL-2.1-only"
+dev-repo: "git+https://git.frama-c.com/pub/frama-c.git"
+doc: "http://frama-c.com/download/user-manual-27.0-Cobalt.pdf"
+bug-reports: "https://git.frama-c.com/pub/frama-c/issues"
+tags: [
+  "deductive"
+  "program verification"
+  "formal specification"
+  "automated theorem prover"
+  "interactive theorem prover"
+  "C"
+  "plugins"
+  "abstract interpretation"
+  "slicing"
+  "weakest precondition"
+  "ACSL"
+  "dataflow analysis"
+  "runtime verification"
+]
+
+build: [
+  ["bash" "dev/disable-plugins.sh" "e-acsl"] { os-family = "windows" }
+  ["dune" "build" "-j%{jobs}%" "--release" "--promote-install-files=false"
+   "@install"
+   "@doc" { with-doc }
+  ]
+]
+
+install: [
+  [make
+   "PREFIX=%{prefix}%" "MANDIR=%{man}%"
+   "DOCDIR=%{doc}%" { with-doc }
+   "install"
+  ]
+]
+
+remove: [
+  [make "PREFIX=%{prefix}%" "-f" "ivette/Makefile.installation" "uninstall"]
+]
+
+run-test: [
+  ["dune" "exec" "--" "frama-c-ptests" "tests" "src/plugins/*/tests"
+  ] { arch != "ppc64" & arch != "x86_32" & arch != "arm32" & os != "macos" }
+  ["dune" "build" "-j%{jobs}%" "@ptests_config"
+  ] { arch != "ppc64" & arch != "x86_32" & arch != "arm32" & os != "macos" }
+]
+
+depends: [
+  "dune" { >= "3.2.0" | (>= "3.5.0" & os="macos") }
+  "dune-configurator"
+  "dune-site"
+
+  ( "alt-ergo-free" | "alt-ergo" )
```

Comment:
I think you need a minimum bound on `alt-ergo-free` (see the failing "lower bounds" test in the CI)

Reply:
I believe that the problem is in the alt-ergo-free.2.0.0 opam package itself. In the same docker, when I try to install alt-ergo-free.2.0.0 (and nothing from Frama-C dependencies), I get the same error. 

Is it some kind of trap to convince me that patches are bad? :laughing: 

It is like something strange happens with [this patch](https://github.com/ocaml/opam-repository/blob/master/packages/alt-ergo-free/alt-ergo-free.2.0.0/files/patch_for_changes_in_num_library.patch). It is supposed to add required instructions to detect `num`, but while the compilation instruction indeed has a `@NUMLIB@` to replace:

```
ocamlopt.opt -c -annot -absname -bin-annot -short-paths -strict-sequence -g -inline 100 @NUMLIB@ -I /home/opam/.opam/4.13/lib/zarith ...
```

It seems that during configure, something is missing. Opam generates the following log:

```
# checking for ocamlfind... yes
# ocamlfind found zarith in -I /home/opam/.opam/4.13/lib/zarith
# ocamlfind found camlzip in -I /home/opam/.opam/4.13/lib/camlzip/../zip
# ocamlfind found ocplib-simplex in -I /home/opam/.opam/4.13/lib/ocplib-simplex
```

While when I configure by hand after applying the patch, I get:

```
checking for ocamlfind... yes
ocamlfind found zarith in -I /home/opam/.opam/4.13/lib/zarith
ocamlfind found num in -I /home/opam/.opam/4.13/lib/num
ocamlfind found camlzip in -I /home/opam/.opam/4.13/lib/camlzip/../zip
ocamlfind found ocplib-simplex in -I /home/opam/.opam/4.13/lib/ocplib-simplex
```

Reply:
Ok I don't know what happened but now the build of Alt-Ergo succeeds :thinking: